### ISSUE TITLE
Mixed up order of notifications having the same timestamp

### ISSUE THREAD

**Issue by emchay** [devicehive/devicehive-admin-console#17]
An order of notifications is mixed up when they have exactly the same timestamp. Notification id should also be considered as a secondary sort order.

**Comment by vyrostkov** [comment]
Can you specify what server do you use (net/java)? The admin console preserve the order returned by the server.

**Comment by emchay** [comment]
we use .net server

**Comment by demon-xxi** [comment]
We are trying to avoid incremental I'd counter because of scalability constraints it introduces. Did those notifications with same timestamp came from same device or not? We may increase precision on timestamp to resolve this as they supposed to be sequential from same device. Order of notifications from different devicesdoes not have to be ordered in any special way if they have _exact_ same timestamp. Wdyt?

**Comment by vyrostkov** [comment]
Do you use MongoDB or MS SQL database on the backend?
That's correct, we can not generally rely on IDs because in MongoDB they are not sequential. However, MongoDB only supports 3-digit precision for timestamps, so there's no way to increase it.

I suppose the problem applies to notifications from the same device, as there's no screen in admin console that combines notification from multiple devices..

**Comment by emchay** [comment]
Yes you are right, the notifications were from a single device. 
And...we use MongoDB unfortunately.
